### ISSUE TITLE
Bug: Read only currency field with negative value displays positive value

### ISSUE THREAD

**Issue by jonom** [silverstripe/silverstripe-framework#8126]
## Affected Version

SS 4.1.0 (Have not checked other versions)

## Steps to Reproduce

```php
$fields->addFieldToTab('Root.Price', CurrencyField::create('CurrencyEdit', null, -1.20));
$fields->addFieldToTab('Root.Price', CurrencyField::create('CurrencyReadOnly', null, -1.20)->performReadOnlyTransformation());
```

### Result:
![image](https://user-images.githubusercontent.com/1079425/40860015-9cbf71d6-6598-11e8-8b15-b495e7a117d1.png)

**Comment by jonom** [comment]
This affects disabled currency fields too

**Comment by dhensby** [comment]
fixed with #8127
